pkg/sql/plan2/function/operator/cast.go

```diff
@@ -1305,6 +1305,9 @@ func castTimestampAsVarchar(lv, rv *vector.Vector, proc *process.Process) (*vect
 	resultType := rv.Typ
 	resultElementSize := int(resultType.Size)
 	if lv.IsScalar() {
```

Comment:
I think you can get the timestamp's precision from lv.Typ.Precision

Reply:
I will add it in a new pr.